Core/include/Acts/TrackFitting/BetheHeitlerApprox.hpp

```diff
@@ -0,0 +1,322 @@
+
+namespace detail {
+
+struct GaussianComponent {
+  ActsScalar weight, mean, var;
+};
+
+/// Transform a gaussian component to a space where all values are defined from
+/// [-inf, inf]
+void transformComponent(const GaussianComponent &cmp,
+                        double &transformed_weight, double &transformed_mean,
+                        double &transformed_var) {
+  const auto &[weight, mean, var] = cmp;
+
+  transformed_weight = std::log(weight) - std::log(1 - weight);
+  transformed_mean = std::log(mean) - std::log(1 - mean);
+  transformed_var = std::log(var);
+}
+
+/// Transform a gaussian component back from the [-inf, inf]-space to the usual
+/// space
+auto inverseTransformComponent(double transformed_weight,
+                               double transformed_mean,
+                               double transformed_var) {
+  GaussianComponent cmp;
+  cmp.weight = 1. / (1 + std::exp(-transformed_weight));
+  cmp.mean = 1. / (1 + std::exp(-transformed_mean));
+  cmp.var = std::exp(transformed_var);
+
+  return cmp;
+}
+
+}  // namespace detail
+
+namespace Experimental {
+
+/// This class approximates the Bethe-Heitler with only one component. The
+struct BetheHeitlerApproxSingleCmp {
+  /// Returns the number of components the returned mixture will have
+  constexpr auto numComponents() const { return 1; }
+
+  /// Checks if an input is valid for the parameterization. Since this is for
+  /// debugging, it always returns false
+  ///
+  /// @param x input in terms of x/x0
+  constexpr bool validXOverX0(ActsScalar) const { return false; }
+
+  /// Returns array with length 1 containing a 1-component-representation of the
+  /// Bethe-Heitler-Distribution
+  static auto mixture(const ActsScalar x) {
+    std::array<detail::GaussianComponent, 1> ret{};
+
+    ret[0].weight = 1.0;
+
+    const double c = x / std::log(2);
+    ret[0].mean = std::pow(2, -c);
+    ret[0].var = std::pow(3, -c) - std::pow(4, -c);
+
+    // ret[0].mean = std::exp(-1. * x);
+    // ret[0].var =
+    //     std::exp(-1. * x * std::log(3.) / std::log(2.)) - std::exp(-2. * x);
+
+    return ret;
+  }
+};
+
+/// This class approximates the Bethe-Heitler distribution as a gaussian
+/// mixture. To enable an approximation for continous input variables, the
+/// weights, means and variances are internally parametrized as a Nth order
+/// polynomial.
+template <int NComponents, int PolyDegree>
+class AtlasBetheHeitlerApprox {
+  static_assert(NComponents > 0);
+  static_assert(PolyDegree > 0);
+
+ public:
+  struct PolyData {
+    std::array<ActsScalar, PolyDegree + 1> weightCoeffs, meanCoeffs, varCoeffs;
+  };
+
+  using Data = std::array<PolyData, NComponents>;
+
+  constexpr static double noChangeLimit = 0.0001;
+  constexpr static double singleGaussianLimit = 0.002;
+  constexpr static double lowerLimit = 0.10;
+  constexpr static double higherLimit = 0.20;
+
+ private:
+  Data m_low_data;
+  Data m_high_data;
+  bool m_low_transform;
+  bool m_high_transform;
```

Comment:
should we default those?

Reply:
Hmm in this case I don't think, because there is no default constructor, and e.g., for the data it is not possible to specify defaults.
But I could explicitly delete the default constructor do be more explicit here.